### ISSUE TITLE
Inconsistent handling of invalid exact answers between train and test set

### ISSUE THREAD

**Issue by LennartBuerger** [technion-cs-nlp/LLMsKnow#10]
Hey! If I understand the code correctly, in the test set passed to the probe function in probe.py (lines 77-80), model answers with an invalid exact answer are filtered out. In contrast, in the training set, model responses with an invalid exact answer are not filtered out. Instead, the probe is trained on the activations over the last question token (line 186 in probing_utils.py). So invalid exact answers are handled differently for train and test set. Fixing this will probably not change the results by much but I still wanted to mention it here. Overall very nice and clean code though! Edit: The training data is also filtered using the compile_probing_indices function and I missed this. So there is no inconsistency.
